File: javascript/hamming/hamming.js
export default class Hamming {
  compute(strand1, strand2) {
    if (strand1.length != strand2.length) {
      throw new Error('DNA strands must be of equal length.')
    }

    if (strand1 == strand2) {
      return 0
    }

    const diff = strand1.split('').filter((dna, i) => dna != strand2[i])
    return diff.length
  }
}

File: ruby/hamming/hamming.rb
class Hamming
  def self.compute (str1, str2)
    raise ArgumentError if str1.length != str2.length
    return 0 if str1 == str2
    ar = []
    str1.split('').each_with_index { |val, i| ar.push(val) if val != str2[i] }
    ar.size
  end
end